presto-main/src/main/java/com/facebook/presto/operator/scalar/ReflectionParametricScalar.java

```diff
@@ -597,7 +654,6 @@ public MethodHandleAndConstructor specialize(Signature boundSignature, BoundVari
             if (nullable) {
                 return Primitives.wrap(clazz);
             }
-            checkArgument(clazz != void.class);
```

Comment:
Oh...
And there is that for `void.class`:
https://github.com/prestodb/presto/blob/c7bf63a2138c2e61f0edb307cefbf6b35fba25c2/presto-main/src/main/java/com/facebook/presto/operator/scalar/FailureFunction.java
It wasn't failing before as the `void.class` check was done in a wrong place and didn't cover all flows.
I think this check was strictly broken just dead for cases it mattered for.

I removed that check once again. I hope you don't mind that.